### PR TITLE
Url rename

### PR DESCRIPTION
**Pull request by michelbl**
Connected to #521

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # Changelog
 
+## 14.0.0
+
+#### Breaking changes - [#525](https://github.com/openfisca/openfisca-core/pull/525)
+
+* Rename Variable attribute `url` to `reference`
+  - As previously, this attribute can be a string, or a list of strings.
+  - Variable attribute `law_reference` is removed.
```
Comment (fpagnoux) [inline CHANGELOG.md]: This should be a different item of the list, at the same level than "Rename...", and at the imperative mode.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+  - `TaxBenefitSystem.reference` renamed to `TaxBenefitSystem.reference_tbs`.
```
Comment (fpagnoux) [inline CHANGELOG.md]: Not sure how I feel about that. I understand how you came to do this renaming (to be consistent about what a `reference` means), but this is not directly related to our issue.


@MattiSG thoughts ?

Comment (fpagnoux) [inline CHANGELOG.md]: If this change is kept: 
    - we should avoid acronyms and call it reference_tax_benefit_system if need be.
    - This is a proper breaking change, and should be a first level item

File: openfisca_core/taxbenefitsystems.py [hunk 3]
```diff
@@ -134,7 +134,7 @@ def load_variable(self, variable_class, update = False):
         existing_column = self.get_column(name)
         if existing_column:
             if update:
-                attributes['reference'] = existing_column
+                attributes['reference_column'] = existing_column
```
Comment (fpagnoux) [inline openfisca_core/taxbenefitsystems.py]: >This implementation allows the formula writer to define a `reference_column` attribute, that will be... ignored in most cases (when we are indeed replacing an existing variable).
>
>This problem existed before this PR, so not sure whether this should be treated here. I'll let you decide :)

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
+  - Parameter `reference` of `AbstractScenario.new_simulation()` renamed to `use_reference_tbs`.
```
Comment (fpagnoux) [inline CHANGELOG.md]: The two `reference` options you're renaming to `use_reference_tbs` have always sounded weird to me: `reform.legislation_at(x, reference=True)` returns the legislation of the tax benefit system the reform was built on 😕 .
I don't know if it's relevant to rename something we may just want to depreciate one of these days 🤔 .

Comment (fpagnoux) [inline CHANGELOG.md]: If this change is kept: 
    - we should avoid acronyms and call it reference_tax_benefit_system if need be.
    - This is a proper breaking change, and should be a first level item

File: CHANGELOG.md (hunk 2 above)
Comment (benjello) [inline CHANGELOG.md]: I would definitively prefer to use legislative_reference` instead of  `reference` and keep `reference` for the dichotomy `reference`/`reform`.

Comment (MattiSG) [inline CHANGELOG.md]: Calling the original version of a variable in a reform “reference” doesn't sound very obvious to me. And I can't find a better alternative to “legislative reference”. Plus, if you consider how many times the current “reference” is used vs how many times “legislative reference” should be used (once for each variable), we should definitely prioritise the most common meaning.

I'm pretty sure we can find an acceptable alternate word for the “reference” tax benefit system in the context of a reform, which would solve both the `TaxBenefitSystem.reference` and `some_variable.reference` attribute in a reform.

What about the “base” tax benefit system? The “original”?

If none of these work out (please explain why), can we call the legislative reference `source`?

Comment (benjello) [inline CHANGELOG.md]: My point was to keep `reference` as an attribute for `TaxBenefitSystem` since their `reference` is natural for two reasons:
 - the reference is also a TaxBenefitSystem
 - `reference/reform` is very clear dichotomy to me (`base` calls for `extension` and `original` calls for `copy`) 
   
Extension of the `reference` to the Variable as `reference_variable` or `reference` is on the same vein.

I am ok to use `source` or `legislative_reference`  instead of  `url`.

Comment (michelbl) [inline CHANGELOG.md]: As you wish. All these variables are now disambiguated so a renaming is straightforward.

However I cannot work on this issue after tomorrow, so if a consensus is not reached yet, someone else will have to make it and merge to master.

**Comment by benjello** [comment]
Quelques résultats de la consultation (encore en cours) à l'IPP  (@SophieIPP , @bfabre01 , @abozio )
 - pour le `TaxBenefitSystem` de référence, on peut utiliser `baseline`
 - par contre `reference` pour la référence législative n'est pas forcément aussi adaptée pour un anglophone que pour un francophone.  Pas de solution géniale trouvée mais `legal_reference`, `law` ont été proposées.

**Comment by MattiSG** [comment]
I really like `baseline` for the tax benefit system of a reform, I find this quite explicit!

`reference` not being obvious for a native English speaker sounds more surprising to me. Academic papers use a “references” chapter to list official references. `law` or `legal` is good for most cases, but might be a bit restrictive as I'm not 100% sure we want to have only law. Sometimes we can refer to interpretation documents too, don't we?

**Comment by MattiSG** [comment]
I've tried something: https://twitter.com/openfisca/status/877806466585042944 #LazyUX 😬

**Comment by benjello** [comment]
Yes different types of document describes how a tax should be computed or the value of some legislation parameters.

**Comment by bfabre01** [comment]
I think "baseline" is indeed more appropriate. It may be better for english speakers. Moreover, what we call "reference" may not correspond to the actual "law" or "legal" system. For instance, when we run simulations on social benefits reform, the "reference" can be the current system with full take-up, while the take-up is partial in the facts.

**Comment by benjello** [comment]
@bfabre01 : be careful `reference` that could be `law` or `legal` is a potential substitue for the `url` attribute of `Variable` not the attribute `reference`  of `TaxBenefitSystem` !

**Comment by MattiSG** [comment]
Let's proceed with:

- `reference` for variable legal references.
- `baseline` for tax and benefit system references.
- `variable_baseline` for variable baseline tax and benefit system references.

capture d ecran 2017-06-22 a 12 23 27

**Comment by MattiSG** [comment]
This wasn't shipped in the current sprint, so we have more time to discuss the naming scheme. Let's think in whole name sets.

**Comment by benjello** [comment]
I am ok for:
- `reference` for variable legal references.
- `baseline` for tax and benefit system references.
- `baseline_variable` for baseline variable

**Comment by MattiSG** [comment]
There is also `use_reference_tbs` (a boolean option, that was before this PR `reference`, triggering the load of the `reference_tbs`). I would go with `use_baseline`.
